CMakeLists.txt

```diff
@@ -89,6 +89,9 @@ if (NOT DEFINED CMAKE_OPTIMIZE_DEPENDENCIES)
   set(CMAKE_OPTIMIZE_DEPENDENCIES TRUE)
 endif()
 
+# Enable export of `compile_commands.json` used by the clangd language server.
+set(CMAKE_EXPORT_COMPILE_COMMANDS ON)
```

Comment:
Variables that start with `CMAKE_` are not intended to be set by projects. IDEs may set `CMAKE_EXPORT_COMPILE_COMMANDS` when configuring the project. You may also set `CMAKE_EXPORT_COMPILE_COMMANDS` as an environment variable in your dotfiles. But don't force it on other developers that have their own needs.

Reply:
I think considerations of the intended semantics of the cmake language are secondary to considerations of what is useful to the project.

Reply:
For the CI, it is enabled externally:https://github.com/bitcoin/bitcoin/blob/57e8f34fe20620b3350ca1f0fa5c7d8d3a06be82/ci/test/03_test_script.sh#L121-L123